editor/code_editor.cpp

```diff
@@ -419,6 +419,34 @@ void FindReplaceBar::_update_matches_label() {
 	}
 }
 
+void FindReplaceBar::_temp_unfold_line(int p_line) {
+	// Unfold p_line, then walk back to find which was the first line that had to be unfolded so p_line can be shown.
+	// Mark that line as temporarily unfolded, so it's folded back on the next temporarily unfolding - unless the user for example edits the text.
+	if (temp_unfolded_line != -1) {
+		text_editor->fold_line(temp_unfolded_line);
+	}
+
+	if (!text_editor->is_line_folded(p_line)) {
+		temp_unfolded_line = p_line;
+		while (temp_unfolded_line >= 0) {
+			if (text_editor->is_line_folded(temp_unfolded_line)) {
+				break;
+			} else {
+				String line_text = text_editor->get_line(temp_unfolded_line);
+				// If at the start of a lowest-level foldable block, don't backtrack further.
+				// TODO This should likely be refined in its own CodeEdit function, but when this was written such a function didn't exist.
+				if (text_editor->can_fold_line(temp_unfolded_line) && !is_whitespace(line_text[0]) && text_editor->is_in_comment(temp_unfolded_line, 0) != -1 && text_editor->is_in_string(temp_unfolded_line, 0) != -1) {
```

Comment:
This now checks that it *is* in both a comment *and* a string

Reply:
Blah, I hoped you wouldn't show up in time to see the blunder x_x

Reply:
Aw I'm sorry, it can be hard